src/ameba.ts

```diff
@@ -29,28 +30,34 @@ export class Ameba {
         this.config = getConfig();
     }
 
-    public execute(document: TextDocument, virtual: boolean = false): void {
-        if (!isValidCrystalDocument(document)) return;
-        if (isDocumentVirtual(document) && !virtual) return;
+    public execute(document: TextDocument | WorkspaceFolder, virtual: boolean = false): void {
+        if (!this.isTextDocument(document)) {
+            virtual = false;
+       } else {
+            if (!isValidCrystalDocument(document)) return;
+            if (isDocumentVirtual(document) && !virtual) return;
+        }
 
         const dir = (workspace.getWorkspaceFolder(document.uri) ?? noWorkspaceFolder(document.uri)).uri.fsPath;
 
         const args = [this.config.command, '--format', 'json'];
         const configFile = path.join(dir, this.config.configFileName);
         if (existsSync(configFile)) args.push('--config', configFile);
 
-        if (!virtual) {
-            args.push(document.fileName)
-        } else {
-            // Disabling these as they're common when typing
-            args.push('--except', 'Lint/Formatting,Layout/TrailingBlankLines,Layout/TrailingWhitespace');
-
-            // Indicate that the source is passed through STDIN
-            if (document.uri.scheme === 'untitled') {
-                args.push('-')
+        if (this.isTextDocument(document)) {
+            if (!virtual) {
+                args.push(document.fileName)
             } else {
-                // Necessary to support excludes in ameba config
-                args.push('--stdin-filename', document.fileName);
+                // Disabling these as they're common when typing
+                args.push('--except', 'Lint/Formatting,Layout/TrailingBlankLines,Layout/TrailingWhitespace');
```

Comment:
should this be configurable?

Reply:
By configurable, do you mean "add more to the list" or "replace all items in the list"?

Reply:
IMO I don't think having a config option for these is worthwhile, but if you think it is, I think adding it is outside of the scope of this PR.

Reply:
Hard to say if it is really needed/required, but I am thinking about some future rules, which could be added to ameba's ruleset that might also have to be disabled when typing.